File: c/test_string.c
// test_string.c
//                           wookay.noh at gmail.com

#include <stdio.h>

void assert_equal(const char* expected, const char* got) {
  if (0 == strcmp(expected, got)) {
    printf("passed: %s\n", expected);
  } else {
    printf("Assertion failed\nExpected: %s\nGot: %s\n", expected, got);
  }
}

int main(int argc, char* argv[]) {

  assert_equal( "a" , "a" );
  return 0;

}

File: java/TestString.java
// TestString.java
//                           wookay.noh at gmail.com

class TestString {
  public static final String QUOTE = "\"";

  public static String inspect(String obj) {
    return QUOTE + obj + QUOTE;
  }

  public static String ljust(String str, int padding) {
    StringBuffer buffer = new StringBuffer(str);
    int idx;
    for (idx = str.length() ; idx < padding; idx++) {
      buffer.append(" ");
    }
    return buffer.toString();
  }

  static void assert_equal(String expected, String got) {
    if (expected.equals(got)) {
      System.out.println("passed: " + inspect(expected));
    } else {
      System.out.println("Assertion failed\nExpected: " + inspect(expected) + "\nGot: " + inspect(got));
    }
  }



  public static void main(String[] args) {
   
    assert_equal( "a    " ,  ljust("a", 5)   );
    assert_equal( "     " ,  ljust("", 5)    );
    assert_equal( ""      ,  ljust("", 0)    );
    assert_equal( "abc"   ,  ljust("abc", 2) );

  }
}
